client/state/login/selectors.js

```diff
@@ -58,7 +52,7 @@ export const getTwoFactorPushToken = ( state ) =>
  * @returns {boolean}         Whether the polling for push authentication is in progress
  */
 export const getTwoFactorPushPollInProgress = ( state ) =>
-	get( state, 'login.twoFactorAuthPushPoll.inProgress', false );
+	state.login.twoFactorAuthPushPoll?.inProgress ?? false;
```

Comment:
I agree that this can and should be simplified, good catch! 👍 

However, this does mean that we now need to add this default state to the tests, or otherwise they will fail.

@tyxla @jsnajdr: What do you think is the correct approach?
- Make the tests depend on the reducer, so that we can use it to prepare the empty state that we test against?
- Keep the tests independent from the reducer, and hardcode the default values into the empty state in the test, e.g.:
  ```js
  const EMPTY_STATE = {
    login: {
      twoFactorAuthPushPoll: {
        inProgress: false,
        success: false,
      },
    },
  };
  ```

Reply:
> However, this does mean that we now need to add this default state to the tests,

I'd say yes. Tests are supposed to mirror the original execution flow, and if we need to be more specific when defining test fixtures, we should go that way.

> Make the selector tests depend on the reducer, so that we can use it to prepare the empty state that we test against?

I'd pick that one. I don't have any reservations against this approach, and have done it before without problems.

Reply:
> Make the selector tests depend on the reducer, so that we can use it to prepare the empty state that we test against?

When reviewing the selectors and deciding whether a particular optional chaining or fallback is needed, I used the reducer code as the source of truth for that. Sometimes the actions are relevant, too. That means they are all part and parcel of one module and it's OK to test them as one unit.

When writing new Redux-related tests, I usually try to create a simple Redux store with the tested reducer and use the actions and selectors on this store. Then it feels that the unit tests do something that's close to a real-world app.

Reply:
Thank you both for the feedback! 👍 

I went with the simpler approach of calling the reducer with an `undefined` previous state and a no-op action. It's not as extensive as creating a store, but I think it fits the model of the selector tests well. Let me know if you disagree!

Reply:
> I went with the simpler approach of calling the reducer with an undefined previous state and a no-op action.

I think that's perfectly fine. By calling `newState = reducer( oldState, action )`, you implemented almost the entire Redux (without subscriptions) on one line 🙂 

The point is that you can create realistic test states just by calling the reducer with an action, instead of maintaining large testing objects and having to modify them on every state shape change.